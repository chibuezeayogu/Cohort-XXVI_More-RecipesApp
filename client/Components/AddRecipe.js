import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Input, Icon, Button, Col, Table} from 'react-materialize';
import axios from 'axios';
import Preloader from './Preloder';
//require('dotenv').config();


const CLOUDINARY_URL = process.env.CLOUDINARY_URL;
const CLOUDINARY_UPLOAD_PRESET = process.env.CLOUDINARY_UPLOAD_PRESET;


class AddRecipe extends Component{

    constructor(props) {
        super(props);
        this.state = {
            file: '',
            imagePreviewUrl: '', 
            ImageURL: '',
            arrIngredients:[],
            arrProcedures:[]
        };
      }

      componentWillMount(){
        // if(!this.props.userData.isAuthenticated){
        //     this.context.router.push('/sign');
        // }
        const token = localStorage.getItem('jwtToken');
        if(!token){
            this.context.router.push('/signin');
        }
    }
    //   componentWillUpdate(nextProps, nextState) {
    //     console.log(nextState.arrIngredients);
    // }
   
      _addIngredients(e){
          e.preventDefault();
          const ingredients = document.getElementById('ingredients').value;
          if(!ingredients){
                alert("Ingredients field is empty");
          }else {

            const newIngredients = this.state.arrIngredients.concat(ingredients);
                this.setState({
                    arrIngredients: newIngredients
                });
            }
      }

     



      _addProcedures(e){
        e.preventDefault();
        const procedures = document.getElementById('procedures').value;
        if(!procedures){
              alert("Ingredients field is empty");
        }else {

          const newProcedures = this.state.arrProcedures.concat(procedures);
              this.setState({
                    arrProcedures: newProcedures
              })
          }
        
          this.ingredient.state.reset();
      }

      _handleImageChange(e) {
        e.preventDefault();
    
        let reader = new FileReader();
        let file = e.target.files[0];
        const formData = new FormData;
        formData.append('file', file);
        formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);
        
        // if (!file.type.match('image.*')){

        // }else{

        // }
        
        $.ajax({
            type: "POST",
            url: CLOUDINARY_URL,
            dataType: 'json',
            data: formData,
            contentType:false,
            processData: false,
            success: ((data) => {
                this.setState({ImageURL:data.secure_url});
            })
        });

        reader.onloadend = () => {
          this.setState({
            file: file,
            imagePreviewUrl: reader.result
          });
        };
    
        reader.readAsDataURL(file);
      }
      
    handleOnsubmit(e) {
        e.preventDefault();
        const title = this.title.state.value;
        const description = this.description.state.value;
        const ingredients = this.ingredients.state.value;
        const procedures = this.procedures.state.value;
        const imageURL = this.state.ImageURL;
        this.props.addRecipe(title, description, ingredients, procedures, imageURL);
        this.refs.AddRecipeForm.reset();
        this.setState({
            imagePreviewUrl:''
        });
    }
    render() {
        // let {imagePreviewUrl} = this.state;
        // let $imagePreview = null;
        // if (imagePreviewUrl) {
        //   $imagePreview =(<img  class="responsive-img circle" src={imagePreviewUrl}/>);
        // } else {
        //   $imagePreview = " ";
        // }
       const  { arrIngredients, arrProcedures } = this.state;
        return (
            <div className="container"> 
                <h1 className="center">Add Recipe</h1>
                <hr/>
                <form encType="multipart/form-data" methos="post" className="col l6 m8 s12 offset-l8 offset-m6" onSubmit={(e) => this.handleOnsubmit(e)} ref="AddRecipeForm">
                    <Row>
                        <Input s={12} label="Title"  type='text' ref={(input)=> {this.title=input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row s={8}>
                        <Input s={12} label="Description" className='validate' type='text' ref={(input) => {this.description = input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row>
                        <Input s={12} label="Ingredients" id="ingredients" type='text' ref={(input) => {this.ingredients = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addIngredients(e)}}>Add Ingredient</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrIngredients
                                    .map((ingredient, i) => 
                                        <tr>
                                            <td className="s10">
                                                <input id="ingredients" type='text' value={ingredient}></input>
                                            </td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={12} label="Procedures" id="procedures" type='text' ref={(input) => {this.procedures = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addProcedures(e)}}>Add Procedures</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrProcedures
                                    .map((procedures, i) => 
                                        <tr>
                                            <td className="s10">{procedures}</td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={6} id="recipe"  name="file" type='file' ref={(input) => {this.file = input}}   onChange={(e)=>this._handleImageChange(e)}><Icon>insert_photo</Icon></Input>
                        <Col s={6} ><img className="responsive-img img-wh z-depth-3" src={this.state.imagePreviewUrl && this.state.ImageURL?this.state.imagePreviewUrl:"select image to preview"}/></Col>
                    </Row>
                    <br />
                    <Row>
                        <Button waves='light' className='right'><Icon right>send</Icon>Add</Button>
                    </Row>
                </form>
            </div>
        );
    }
}

AddRecipe.contextTypes = {
    router: React.PropTypes.object
};


AddRecipe.propTypes = {
handleOnSubmit: PropTypes.func.isRequired,
_addIngredients:  PropTypes.func.isRequired,
_addProcedures:  PropTypes.func.isRequired,
_handleImageChange:  PropTypes.func.isRequired,
addRecipe: PropTypes.func.isRequired,
};

export default AddRecipe;