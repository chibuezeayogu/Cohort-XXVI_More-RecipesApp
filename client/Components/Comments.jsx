import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const Comments = props => (
  <li className="collection-item avatar z-depth-1">
    <img
      src={props.review.User.imageUrl}
      alt="profile"
      className="circle responsive-img"
    />
    <span className="text-title bold">{props.review.User.firstName}</span>
    <p
      style={{ wordWrap: 'break-word' }}
    >
      {props.review.comment}
    </p>
    <div className="secondary-content">
      { moment(new Date(props.review.createdAt)).fromNow()}
    </div>
  </li>
);

Comments.propTypes = {
  review: PropTypes.shape({
    comment: PropTypes.string,
    createdAt: PropTypes.string,
    User: PropTypes.shape({
      firstName: PropTypes.string,
      imageUrl: PropTypes.string
    })
  }).isRequired
};

export default Comments;
