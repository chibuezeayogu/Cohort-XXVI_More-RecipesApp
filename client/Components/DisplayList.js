import React, { Component } from 'react';
import { Link } from 'react-router';

const Displaylist = React.createClass ({
    render(){
        const { recipes } = this.props;
        return(
            <td>
                list ingradients
            </td>
        );
    }
});

export default Displaylist;