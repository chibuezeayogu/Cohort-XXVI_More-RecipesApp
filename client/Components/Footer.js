import React, { Component }from 'react';
import PropTypes from 'prop-types';
import { Footer } from 'react-materialize';


class BottomFooter extends  Component {
    render() {
        return (
            <div>
                <Footer copyrights="&copy 2017 Copyright Text"
                moreLinks={
                    <a className="grey-text text-lighten-4 right" href="#!">Andela Project by Ayogu Chibueze></a>
                }
                
                className='black'
            >
                    
                </Footer>;
            </div>
        );
    }
}

BottomFooter.propTypes = {
    logout: PropTypes.func.isRequired,
    history: PropTypes.func.isRequired,
    push: PropTypes.func.isRequired,
    className: PropTypes.string,
    copyrights: PropTypes.string,
};

export default BottomFooter;