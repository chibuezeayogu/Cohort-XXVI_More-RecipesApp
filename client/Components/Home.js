import React, { Component } from 'react';
import { Row, Input, Icon, Textarea, Slider, Slide, Parallax } from 'react-materialize';
import NavBar from  './NavBar';
import Footer from  './Footer';


class Home extends Component {
    render() {
        return (
            <div>
                <Slider className="full-screen slide-margin-top">
                    <Slide
                        src="https://static.pexels.com/photos/161640/abstract-barbecue-barbeque-bbq-161640.jpeg"
                        title="More Recipes">
                        More-Recipes provides a platform for users to share the awesome and exciting recipe ideas they have invented or learnt.
                    </Slide>
                    <Slide
                        src="https://static.pexels.com/photos/213621/pexels-photo-213621.jpeg"
                        title="Easy to Setup"
                        placement="left">
                        npm install react-materialize.
                    </Slide>
                    <Slide
                        src="https://static.pexels.com/photos/51423/bun-chocolate-sweets-dessert-51423.jpeg"
                        title="Unlimited Oppurtunities"
                        placement="right">
                        Contribute and Help to Grow this Project.
                    </Slide>
                </Slider>
            </div>
        );
    }
}

export default Home;