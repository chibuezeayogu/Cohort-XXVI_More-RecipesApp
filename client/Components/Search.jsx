import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { search, fetchRecipesWithMostUpvote } from '../action/actionCreators';
import UserMenu from './Header/UserMenu.jsx';
import Footer from './Footer/Footer.jsx';
import SearchResult from './SearchResult.jsx';
import Pagination from 'rc-pagination';
import Preloader from './Preloder.jsx';

export class Search extends Component {
  constructor() {
    super();
    this.state = ({ isLoading: false });
    this.handleSearch = this.handleSearch.bind(this);
  }

  componentWillMount() {
    this.state = ({ isLoading: true });
    this.props.fetchRecipesWithMostUpvote();
  }
  /**
   * @description checks if the there is a new props
   *
   * @method
   *
   * @memberOf Search
   *
   * @param {Object} nextProps - nextProps object
   *
   * @returns {undefined}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.searchReducer.isFetched) {
      this.setState({ isLoading: false });
    }
  }

  handleSearch(event) {
    event.preventDefault();
    const serchTerm = event.target.value;
    if ((event.target.value).length > 2) {
      this.props.search(serchTerm);
      this.state = ({ isLoading: true });
    } else {
      this.props.fetchRecipesWithMostUpvote();
    }
  }

  /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf Search
   *
   * @returns {undefined}
   *
   */
  render() {
    const { recipes, display, pagination } = this.props.searchReducer;
    return (
      <div className="body">
        <UserMenu {...this.props} />
        <div className="main">
          <div className="container">
            <div className="row">
              <h4 className="center">Search Recipe</h4>
              <hr />
            </div>
            <div className="row">
              <form
                className="col s12"
                onChange={this.handleSearch}
              >
                <div className="row">
                  <div className="input-field col s12">
                    <input name="searchTerm" id="search" type="text" onChange={this.handleSearch} />
                    <label htmlFor="search">Enter Search term</label>
                  </div>
                </div>
              </form>
            </div>
            <h4 className="center">{display}</h4>
            {this.state.isLoading ?
              <Preloader />
            :
              <SearchResult {...this.props} />
            }
          </div>
        </div>
        <div className="row s12 m6 l3">
          { recipes && recipes.length > 0 ?
            <Pagination
              onChange={this.onChange}
              current={pagination.currentPage}
              pageSize={pagination.pageSize}
              total={pagination.totalCount}
            />
          : ''
          }
        </div>
        <Footer />
      </div>
    );
  }
}

Search.propTypes = {
  searchReducer: PropTypes.shape({
    pagination: PropTypes.shape({
      currentPage: PropTypes.number,
      pageSize: PropTypes.number,
      totalCount: PropTypes.number
    }).isRequired,
    recipes: PropTypes.shape.isRequired,
    display: PropTypes.string.isRequired,
    isFetched: PropTypes.bool.isRequired,
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  fetchRecipesWithMostUpvote: PropTypes.func.isRequired,
  search: PropTypes.func.isRequired,
};


const mapStateToProps = state => ({
  userData: state.userData,
  searchReducer: state.searchReducer
});

export default connect(mapStateToProps,
  { search, fetchRecipesWithMostUpvote })(Search);
