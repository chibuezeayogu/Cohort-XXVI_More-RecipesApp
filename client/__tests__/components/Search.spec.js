import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { Search } from '../../Components/Search';
import mockData from '../__mock__/actionMockData';

const props = {
  searchReducer: {
    pagination: mockData.data.pagination,
    recipes: mockData.data.recipes,
    display: 'Top Recipes'
  },
  fetchRecipesWithMostUpvote: jest.fn(),
  search: jest.fn()
};

const event = {
  preventDefault: jest.fn(),
  target: {
    name: 'serchTerm',
    value: 'new recipe',
  }
};
const state = {
  isLoading: false,
};

describe('<Search />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Search {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Search />', () => {
  it('should call componentWillReceiveProps', () => {
    const wrapper = shallow(<Search {...props} {...state} />);
    const componentWillReceivePropsSpy = jest
      .spyOn(wrapper.instance(), 'componentWillReceiveProps');
    wrapper.setProps({
      ...props,
      recipeReducer:
      {
        isFetched: true,
        recipes: mockData.data.recipes,
        pagination: mockData.data.pagination
      }
    });
    wrapper.setState({ isLoading: false });
    expect(wrapper.instance().state.isLoading).toEqual(false);
    expect(componentWillReceivePropsSpy).toHaveBeenCalled();
  });
});

// describe('<Search />', () => {
//   it('should call handleChange function', () => {
//     const wrapper = shallow(<Search {...props} />);
//     wrapper.find('#searchterm').simulate('change', event);
//     expect(wrapper.instance().props.search).toHaveBeenCalled();
//   });
// });
