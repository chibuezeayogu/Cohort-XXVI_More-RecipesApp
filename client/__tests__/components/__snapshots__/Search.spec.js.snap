// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <div
    className="body"
  >
    <withRouter(Connect(UserMenu))
      fetchRecipesWithMostUpvote={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
      search={[MockFunction]}
      searchReducer={
        Object {
          "display": "Top Recipes",
          "pagination": Object {
            "currentPage": 1,
            "pageCount": 1,
            "pageSize": 2,
            "totalCount": 2,
          },
          "recipes": Array [
            Object {
              "addedBy": 1,
              "createdAt": "2018-01-29T11:38:51.147Z",
              "description": "adsfasfa",
              "downvotes": 0,
              "id": 3,
              "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
              "ingredients": "asdfafafa",
              "procedures": "adfadfadf",
              "title": "New title",
              "updatedAt": "2018-02-02T15:30:35.425Z",
              "upvotes": 1,
              "views": 0,
            },
            Object {
              "addedBy": 1,
              "createdAt": "2018-01-28T22:46:03.051Z",
              "description": "asdfafd",
              "downvotes": 0,
              "id": 1,
              "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
              "ingredients": "adfadfadf",
              "procedures": "adfafadf",
              "title": "CHANGED",
              "updatedAt": "2018-02-06T19:28:36.149Z",
              "upvotes": 1,
              "views": 0,
            },
          ],
        }
      }
    />
    <div
      className="main"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <h4
            className="center"
          >
            Search Recipe
          </h4>
          <hr />
        </div>
        <div
          className="row"
        >
          <form
            className="col s12"
            onChange={[Function]}
          >
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <input
                  id="search"
                  name="searchTerm"
                  onChange={[Function]}
                  type="text"
                />
                <label
                  htmlFor="search"
                >
                  Enter Search term
                </label>
              </div>
            </div>
          </form>
        </div>
        <h4
          className="center"
        >
          Top Recipes
        </h4>
        <Preloader />
      </div>
    </div>
    <div
      className="row s12 m6 l3"
    >
      <Pagination
        className=""
        current={1}
        defaultCurrent={1}
        defaultPageSize={10}
        hideOnSinglePage={false}
        itemRender={[Function]}
        locale={
          Object {
            "items_per_page": "条/页",
            "jump_to": "跳至",
            "jump_to_confirm": "确定",
            "next_3": "向后 3 页",
            "next_5": "向后 5 页",
            "next_page": "下一页",
            "page": "页",
            "prev_3": "向前 3 页",
            "prev_5": "向前 5 页",
            "prev_page": "上一页",
          }
        }
        onChange={[Function]}
        onShowSizeChange={[Function]}
        pageSize={2}
        prefixCls="rc-pagination"
        selectComponentClass={null}
        selectPrefixCls="rc-select"
        showLessItems={false}
        showQuickJumper={false}
        showSizeChanger={false}
        showTitle={true}
        style={Object {}}
        total={2}
      />
    </div>
    <Footer />
  </div>,
  "nodes": Array [
    <div
      className="body"
    >
      <withRouter(Connect(UserMenu))
        fetchRecipesWithMostUpvote={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
          }
        }
        search={[MockFunction]}
        searchReducer={
          Object {
            "display": "Top Recipes",
            "pagination": Object {
              "currentPage": 1,
              "pageCount": 1,
              "pageSize": 2,
              "totalCount": 2,
            },
            "recipes": Array [
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-29T11:38:51.147Z",
                "description": "adsfasfa",
                "downvotes": 0,
                "id": 3,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "asdfafafa",
                "procedures": "adfadfadf",
                "title": "New title",
                "updatedAt": "2018-02-02T15:30:35.425Z",
                "upvotes": 1,
                "views": 0,
              },
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-28T22:46:03.051Z",
                "description": "asdfafd",
                "downvotes": 0,
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "adfadfadf",
                "procedures": "adfafadf",
                "title": "CHANGED",
                "updatedAt": "2018-02-06T19:28:36.149Z",
                "upvotes": 1,
                "views": 0,
              },
            ],
          }
        }
      />
      <div
        className="main"
      >
        <div
          className="container"
        >
          <div
            className="row"
          >
            <h4
              className="center"
            >
              Search Recipe
            </h4>
            <hr />
          </div>
          <div
            className="row"
          >
            <form
              className="col s12"
              onChange={[Function]}
            >
              <div
                className="row"
              >
                <div
                  className="input-field col s12"
                >
                  <input
                    id="search"
                    name="searchTerm"
                    onChange={[Function]}
                    type="text"
                  />
                  <label
                    htmlFor="search"
                  >
                    Enter Search term
                  </label>
                </div>
              </div>
            </form>
          </div>
          <h4
            className="center"
          >
            Top Recipes
          </h4>
          <Preloader />
        </div>
      </div>
      <div
        className="row s12 m6 l3"
      >
        <Pagination
          className=""
          current={1}
          defaultCurrent={1}
          defaultPageSize={10}
          hideOnSinglePage={false}
          itemRender={[Function]}
          locale={
            Object {
              "items_per_page": "条/页",
              "jump_to": "跳至",
              "jump_to_confirm": "确定",
              "next_3": "向后 3 页",
              "next_5": "向后 5 页",
              "next_page": "下一页",
              "page": "页",
              "prev_3": "向前 3 页",
              "prev_5": "向前 5 页",
              "prev_page": "上一页",
            }
          }
          onChange={[Function]}
          onShowSizeChange={[Function]}
          pageSize={2}
          prefixCls="rc-pagination"
          selectComponentClass={null}
          selectPrefixCls="rc-select"
          showLessItems={false}
          showQuickJumper={false}
          showSizeChanger={false}
          showTitle={true}
          style={Object {}}
          total={2}
        />
      </div>
      <Footer />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Search
        fetchRecipesWithMostUpvote={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
          }
        }
        search={[MockFunction]}
        searchReducer={
          Object {
            "display": "Top Recipes",
            "pagination": Object {
              "currentPage": 1,
              "pageCount": 1,
              "pageSize": 2,
              "totalCount": 2,
            },
            "recipes": Array [
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-29T11:38:51.147Z",
                "description": "adsfasfa",
                "downvotes": 0,
                "id": 3,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "asdfafafa",
                "procedures": "adfadfadf",
                "title": "New title",
                "updatedAt": "2018-02-02T15:30:35.425Z",
                "upvotes": 1,
                "views": 0,
              },
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-28T22:46:03.051Z",
                "description": "asdfafd",
                "downvotes": 0,
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "adfadfadf",
                "procedures": "adfafadf",
                "title": "CHANGED",
                "updatedAt": "2018-02-06T19:28:36.149Z",
                "upvotes": 1,
                "views": 0,
              },
            ],
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Search {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleSearch": [Function],
        "props": Object {
          "fetchRecipesWithMostUpvote": [MockFunction] {
            "calls": Array [
              Array [],
            ],
          },
          "search": [MockFunction],
          "searchReducer": Object {
            "display": "Top Recipes",
            "pagination": Object {
              "currentPage": 1,
              "pageCount": 1,
              "pageSize": 2,
              "totalCount": 2,
            },
            "recipes": Array [
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-29T11:38:51.147Z",
                "description": "adsfasfa",
                "downvotes": 0,
                "id": 3,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "asdfafafa",
                "procedures": "adfadfadf",
                "title": "New title",
                "updatedAt": "2018-02-02T15:30:35.425Z",
                "upvotes": 1,
                "views": 0,
              },
              Object {
                "addedBy": 1,
                "createdAt": "2018-01-28T22:46:03.051Z",
                "description": "asdfafd",
                "downvotes": 0,
                "id": 1,
                "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                "ingredients": "adfadfadf",
                "procedures": "adfafadf",
                "title": "CHANGED",
                "updatedAt": "2018-02-06T19:28:36.149Z",
                "upvotes": 1,
                "views": 0,
              },
            ],
          },
        },
        "refs": Object {},
        "state": Object {
          "isLoading": true,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="body"
        >
          <withRouter(Connect(UserMenu))
            fetchRecipesWithMostUpvote={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
              }
            }
            search={[MockFunction]}
            searchReducer={
              Object {
                "display": "Top Recipes",
                "pagination": Object {
                  "currentPage": 1,
                  "pageCount": 1,
                  "pageSize": 2,
                  "totalCount": 2,
                },
                "recipes": Array [
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-29T11:38:51.147Z",
                    "description": "adsfasfa",
                    "downvotes": 0,
                    "id": 3,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "asdfafafa",
                    "procedures": "adfadfadf",
                    "title": "New title",
                    "updatedAt": "2018-02-02T15:30:35.425Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-28T22:46:03.051Z",
                    "description": "asdfafd",
                    "downvotes": 0,
                    "id": 1,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "adfadfadf",
                    "procedures": "adfafadf",
                    "title": "CHANGED",
                    "updatedAt": "2018-02-06T19:28:36.149Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                ],
              }
            }
          />
          <div
            className="main"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <h4
                  className="center"
                >
                  Search Recipe
                </h4>
                <hr />
              </div>
              <div
                className="row"
              >
                <form
                  className="col s12"
                  onChange={[Function]}
                >
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <input
                        id="search"
                        name="searchTerm"
                        onChange={[Function]}
                        type="text"
                      />
                      <label
                        htmlFor="search"
                      >
                        Enter Search term
                      </label>
                    </div>
                  </div>
                </form>
              </div>
              <h4
                className="center"
              >
                Top Recipes
              </h4>
              <Preloader />
            </div>
          </div>
          <div
            className="row s12 m6 l3"
          >
            <Pagination
              className=""
              current={1}
              defaultCurrent={1}
              defaultPageSize={10}
              hideOnSinglePage={false}
              itemRender={[Function]}
              locale={
                Object {
                  "items_per_page": "条/页",
                  "jump_to": "跳至",
                  "jump_to_confirm": "确定",
                  "next_3": "向后 3 页",
                  "next_5": "向后 5 页",
                  "next_page": "下一页",
                  "page": "页",
                  "prev_3": "向前 3 页",
                  "prev_5": "向前 5 页",
                  "prev_page": "上一页",
                }
              }
              onChange={[Function]}
              onShowSizeChange={[Function]}
              pageSize={2}
              prefixCls="rc-pagination"
              selectComponentClass={null}
              selectPrefixCls="rc-select"
              showLessItems={false}
              showQuickJumper={false}
              showSizeChanger={false}
              showTitle={true}
              style={Object {}}
              total={2}
            />
          </div>
          <Footer />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="body"
        >
          <withRouter(Connect(UserMenu))
            fetchRecipesWithMostUpvote={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
              }
            }
            search={[MockFunction]}
            searchReducer={
              Object {
                "display": "Top Recipes",
                "pagination": Object {
                  "currentPage": 1,
                  "pageCount": 1,
                  "pageSize": 2,
                  "totalCount": 2,
                },
                "recipes": Array [
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-29T11:38:51.147Z",
                    "description": "adsfasfa",
                    "downvotes": 0,
                    "id": 3,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "asdfafafa",
                    "procedures": "adfadfadf",
                    "title": "New title",
                    "updatedAt": "2018-02-02T15:30:35.425Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                  Object {
                    "addedBy": 1,
                    "createdAt": "2018-01-28T22:46:03.051Z",
                    "description": "asdfafd",
                    "downvotes": 0,
                    "id": 1,
                    "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
                    "ingredients": "adfadfadf",
                    "procedures": "adfafadf",
                    "title": "CHANGED",
                    "updatedAt": "2018-02-06T19:28:36.149Z",
                    "upvotes": 1,
                    "views": 0,
                  },
                ],
              }
            }
          />
          <div
            className="main"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <h4
                  className="center"
                >
                  Search Recipe
                </h4>
                <hr />
              </div>
              <div
                className="row"
              >
                <form
                  className="col s12"
                  onChange={[Function]}
                >
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <input
                        id="search"
                        name="searchTerm"
                        onChange={[Function]}
                        type="text"
                      />
                      <label
                        htmlFor="search"
                      >
                        Enter Search term
                      </label>
                    </div>
                  </div>
                </form>
              </div>
              <h4
                className="center"
              >
                Top Recipes
              </h4>
              <Preloader />
            </div>
          </div>
          <div
            className="row s12 m6 l3"
          >
            <Pagination
              className=""
              current={1}
              defaultCurrent={1}
              defaultPageSize={10}
              hideOnSinglePage={false}
              itemRender={[Function]}
              locale={
                Object {
                  "items_per_page": "条/页",
                  "jump_to": "跳至",
                  "jump_to_confirm": "确定",
                  "next_3": "向后 3 页",
                  "next_5": "向后 5 页",
                  "next_page": "下一页",
                  "page": "页",
                  "prev_3": "向前 3 页",
                  "prev_5": "向前 5 页",
                  "prev_page": "上一页",
                }
              }
              onChange={[Function]}
              onShowSizeChange={[Function]}
              pageSize={2}
              prefixCls="rc-pagination"
              selectComponentClass={null}
              selectPrefixCls="rc-select"
              showLessItems={false}
              showQuickJumper={false}
              showSizeChanger={false}
              showTitle={true}
              style={Object {}}
              total={2}
            />
          </div>
          <Footer />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Search
    fetchRecipesWithMostUpvote={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    search={[MockFunction]}
    searchReducer={
      Object {
        "display": "Top Recipes",
        "pagination": Object {
          "currentPage": 1,
          "pageCount": 1,
          "pageSize": 2,
          "totalCount": 2,
        },
        "recipes": Array [
          Object {
            "addedBy": 1,
            "createdAt": "2018-01-29T11:38:51.147Z",
            "description": "adsfasfa",
            "downvotes": 0,
            "id": 3,
            "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
            "ingredients": "asdfafafa",
            "procedures": "adfadfadf",
            "title": "New title",
            "updatedAt": "2018-02-02T15:30:35.425Z",
            "upvotes": 1,
            "views": 0,
          },
          Object {
            "addedBy": 1,
            "createdAt": "2018-01-28T22:46:03.051Z",
            "description": "asdfafd",
            "downvotes": 0,
            "id": 1,
            "imageUrl": "https://res.cloudinary.com/chibuezeayogu/image/upload/v1517405763/v9wpm63o7fiyktplhfoz.jpg",
            "ingredients": "adfadfadf",
            "procedures": "adfafadf",
            "title": "CHANGED",
            "updatedAt": "2018-02-06T19:28:36.149Z",
            "upvotes": 1,
            "views": 0,
          },
        ],
      }
    }
  />,
}
`;
